llama/generation.py

```diff
@@ -150,21 +150,22 @@ def __init__(self, model: Transformer, tokenizer: Tokenizer, device: torch.devic
                     xs.mark_sharding(layer.attention.cache_k, mesh, (None, None, 0, None))
                     xs.mark_sharding(layer.attention.cache_v, mesh, (None, None, 0, None))
 
+            spmd_mesh = xs.HybridMesh(ici_mesh_shape=(1, num_devices), axis_names=('data', 'model'))
```

Comment:
The mesh seems different from the mesh being used in the kv_cache above.

Reply:
Updated to use the same mesh as above. 

This hybrid mesh was used to do some testing against using the same mesh as kv_cache above. Surprisingly it did slightly (< 10%) better on smaller batch sizes (bs=1, bs=2) but started to do worse on higher batch sizes. However, just updated to use the same mesh that is used for kv_cache in both. Also included the profiler hints to this commit, I'll be updating sheet with the new profile runs.